fair/2.0.0/emissions_driven.py

```diff
@@ -0,0 +1,186 @@
+import pandas as pd
+import numpy as np
+from tools import unifiedtools
+
+def run(inp_df, cfg):
+    """
+    Run FaIR 2.0
+
+    Parameters
+    ----------
+    inp_df : :obj:`pd.DataFrame`
+        Input :obj:`pd.DataFrame` containing the timeseries to run
+
+    cfg : dict
+        Dictionary containing the configuration for this run
+
+    Returns
+    -------
+    :obj:`pd.DataFrame`
+        Results of the run
+    """
+    np_input = _convert_df_to_numpy(inp_df)
+    #res_numpy = _run_numpy(np_input, cfg)
+    #res = _convert_numpy_output_to_df(res_numpy)
+    res = pd.DataFrame()
+    return res
+
+def _run_numpy(inp_ar, 
+                a1,
+                a2,
+                a3,
+                a4,
+                tau1,
+                tau2,
+                tau3,
+                tau4,
+                r0,
+                rC,
+                rT,
+                rA,
+                PI_conc,
+                emis2conc,
+                f1,
+                f2,
+                f3,
+                d,
+                q,
+                ext_forcing,
+                timestep):
+    """
+    Run FaIR 2.0 from numpy array
+
+    This function is internal/private (because its name starts with an
+    underscore) i.e. not intended for use by people other than developers.
+    I wouldn't recommend having a 'public' interface which is just based on
+    numpy as it's too easy to get column orders etc. confused so it'll create
+    more problems than its worth.
+
+    This function can *only* run one scenario, thermal parameter set & gas parameter set at a time
+
+    Parameters
+    ----------
+    inp_ar : :obj:`np.ndarray`
+        Input :obj:`np.ndarray` containing the timeseries to run. No checks
+        of the column order are performed here.
+        format: [[species],[time]]
+
+    a1,...,f3 : :obj:`np.ndarray`
```

Comment:
probably worth listing them in full rather than using ... (to avoid confusion of e.g. is there a b1?)

Reply:
Done - makes sense, keep things simple!